subsys/mgmt/ec_host_cmd/backends/ec_host_cmd_backend_usb.c

```diff
@@ -0,0 +1,674 @@
+/*
+ * Copyright (c) 2025 Google LLC
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <assert.h>
+#include <stdarg.h>
+#include <string.h>
+
+#include <zephyr/device.h>
+#include <zephyr/devicetree.h>
+#include <zephyr/drivers/usb/udc.h>
+#include <zephyr/init.h>
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/mgmt/ec_host_cmd/backend.h>
+#include <zephyr/mgmt/ec_host_cmd/ec_host_cmd.h>
+#include <zephyr/sys/util.h>
+#include <zephyr/usb/usbd.h>
+
+LOG_MODULE_REGISTER(ec_host_cmd_usb, LOG_LEVEL_DBG);
+
+BUILD_ASSERT((CONFIG_EC_HOST_CMD_HANDLER_BUFFER_ALIGN % UDC_BUF_ALIGN) == 0, "Buffers not aligned");
+
+#define USB_SUBCLASS_GOOGLE_EC_HOST_CMD 0x5a
+#define USB_PROTOCOL_GOOGLE_EC_HOST_CMD 0x00
+
+/* Supported version of host commands protocol. */
+#define EC_HOST_REQUEST_VERSION 3
+
+#define EC_HOST_CMD_SEND_TIMEOUT_MS 100
+
+#define EP_BULK_SIZE     64U
+#define EP_INT_SIZE_SIZE 4U
+
+#define INT_POOL_SIZE 4
+
+/* Timeout for receiving entire request. */
+#define OUT_TRANSFER_TIMEOUT_MS 100
+/* Timeout for sending response, starting from notifing host that it is ready. */
+#define IN_TRANSFER_TIMEOUT_MS  200
+
+enum ec_host_cmd_usb_irq_type {
+	USB_EC_HOST_CMD_IRQ_TYPE_EVENT = 0,
+	USB_EC_HOST_CMD_IRQ_TYPE_RESP_READY = 1,
+};
+
+NET_BUF_POOL_DEFINE(ec_host_cmd_ep_pool, 2, 0, sizeof(struct udc_buf_info), NULL);
+UDC_BUF_POOL_DEFINE(int_ep_pool, INT_POOL_SIZE, EP_INT_SIZE_SIZE, sizeof(struct udc_buf_info),
```

Comment:
You can use UDC driver pool, IMHO it is not worth it to define such small pool.
https://github.com/zephyrproject-rtos/zephyr/blob/789f3cea8c17792077774c974d01546896dd80df/subsys/usb/device_next/class/usbd_cdc_acm.c#L600-L604